src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -4,26 +4,71 @@
 /* global Nimiq */
 /* global AmountTooSmallError, NetworkMissmatchError */
 /* global SignTransactionWithPassphrase, SignTransactionWithPin */
+/* global TransactionType */
+/* global AddressUtils */
+/* global Utf8Tools */
 
 class SignTransactionApi extends PopupApi { // eslint-disable-line no-unused-vars
     /**
      * @param {TransactionRequest} txRequest
      */
     async onRequest(txRequest) {
-        if (Nimiq.Policy.coinsToSatoshis(txRequest.value) < 1) {
+        // Validation
+        if (!txRequest) throw new Error('Invalid request');
+
+        // These checks throw an exception on failure
+        AddressUtils.isUserFriendlyAddress(txRequest.sender);
+        AddressUtils.isUserFriendlyAddress(txRequest.recipient);
+        AddressUtils.isUserFriendlyAddress(txRequest.signer);
+        if (typeof txRequest.extraData !== 'string') {
+            throw new Error('Transaction extraData must be a string');
+        }
+
+        // Normalization
+        txRequest.value = Nimiq.Policy.coinsToSatoshis(txRequest.value);
+        txRequest.fee = Nimiq.Policy.coinsToSatoshis(txRequest.fee);
+        txRequest.sender = AddressUtils.formatAddress(txRequest.sender);
+        txRequest.recipient = AddressUtils.formatAddress(txRequest.recipient);
+        txRequest.signer = AddressUtils.formatAddress(txRequest.signer);
+        txRequest.extraData = Utf8Tools.utf8ByteArrayToString(
+            Utf8Tools.stringToUtf8ByteArray(txRequest.extraData || ''),
+        );
+
+        switch (txRequest.type) {
+        case TransactionType.BASIC:
+            if (txRequest.sender !== txRequest.signer) {
+                throw new Error('Sender must be signer for basic transactions');
+            }
+
+            if (txRequest.extraData) {
+                throw new Error('Cannot add extraData to basic transaction');
+            }
+
+            break;
+        case TransactionType.EXTENDED:
+            // Validate extended transactions
```

Comment:
what exactly has to be validated?

Reply:
I couldn't think of any, that's why it's still empty ;)